lib/Migration/Version150004Date20201009183830.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\News\Migration;
+
+use Closure;
+use OCP\DB\ISchemaWrapper;
+use OCP\IDBConnection;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+
+/**
+ * Auto-generated migration step: Please modify to your needs!
+ */
+class Version150004Date20201009183830 extends SimpleMigrationStep {
+
+    protected $connection;
+
+    public function __construct(IDBConnection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 */
+	public function preSchemaChange(IOutput $output, Closure $schemaClosure, array $options) {
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 * @return null|ISchemaWrapper
+	 */
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options) {
+        /** @var ISchemaWrapper $schema */
+        $schema = $schemaClosure();
+
+        if ($schema->hasTable('news_items') &&
+            $schema->getTable('news_items')->hasColumn('feed_id')) {
+            $schema->getTable('news_items')
+                ->getColumn('feed_id')
+                ->setNotnull(true)
+                ->setUnsigned(true);
+        }
+        if ($schema->hasTable('news_feeds') &&
+            $schema->getTable('news_feeds')->hasColumn('folder_id')) {
+            $schema->getTable('news_feeds')
+                ->getColumn('folder_id')
+                ->setUnsigned(true)
+                ->setNotnull(false)
+                ->setDefault(null);
+        }
+
+        return $schema;
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 */
+	public function postSchemaChange(IOutput $output, Closure $schemaClosure, array $options) {
+        $qb = $this->connection->getQueryBuilder();
+
+        $qb->delete('news_items')
+            ->from('news_items', 'i')
+            ->leftJoin('i', 'news_feeds', 'f', 'i.feed_id = f.id')
+            ->where('f.id IS NULL')
+            ->execute();
+
+        $qb->delete('news_feeds')
+            ->from('news_feeds', 'fe')
+            ->leftJoin('fe', 'news_folders', 'fo', 'fe.folder_id = fo.id')
+            ->where('fo.id IS NULL')
+            ->execute();
```

Comment:
This deletes feeds assigned to the root folder, since 0 isn't a folder_id specified in news_folders

Reply:
Oh, DELETE doesn't even seem to support LEFT JOIN